src/components/new-member-list/index.js

```diff
@@ -1,22 +1,42 @@
+/* eslint-disable consistent-return */
+/* eslint-disable array-callback-return */
 import NewMemberListItem from '@components/new-member-list-item';
 import React from 'react';
 import classNames from '@components/new-member-list/new-member-list.module.scss';
 import { membersContext } from '@store/members/members-context';
 
-const NewMemberList = () => {
+const NewMemberList = ({ searchTerm }) => {
   const {
     state: { newMembersArr },
   } = membersContext();
   if (newMembersArr) {
     return (
       <div className={classNames.container}>
-        {newMembersArr.map((ele) => (
-          <React.Fragment key={ele.id}>
-            {(ele.first_name || ele.username) && (
-              <NewMemberListItem newMemberDetails={ele} />
-            )}
-          </React.Fragment>
-        ))}
+        {newMembersArr.length > 0 &&
+          newMembersArr
+            .filter((ele) => {
+              if (searchTerm !== '') {
```

Comment:
Same thought process, repetition of code here which can be handled through util files

Reply:
ok